domaincommon/src/state.rs

```diff
@@ -165,6 +184,38 @@ impl MatchState {
         state.turn_status.taken.insert(mech.to_string());
         state
     }
+
+    fn reduce_mech_actions(state: &MatchState, mech: &str, points_consumed: &u32) -> MatchState {
```

Comment:
Likely just my own warped perspective, but when I see `reduce` here I'm expecting to see something like the reducer pattern from react/functional languages. Might be slightly reasier to read if this was `deduct_mech_action_points` ?